bitnami/external-dns/templates/_helpers.tpl

```diff
@@ -663,6 +668,19 @@ external-dns: linode.apiToken
 {{- end -}}
 {{- end -}}
 
+{{/*
+Validate values of External DNS:
+- must provide the NS1 API key when provider is "ns1"
+*/}}
+{{- define "external-dns.validateValues.ns1.apiKey" -}}
+{{- if and (eq .Values.provider "ns1") (not .Values.ns1.apiKey) (not .Values.ns1.secretName) -}}
+external-dns: ns1.apiKey
+    You must provide the a NS API key when provider="ns1".
+    Please set the token parameter (--set ns1.apiKey="xxxx")
+    or specify a secret that contains an API key. (--set ns1.secretName="xxxx")
+{{- end -}}
+{{- end -}}
```

Comment:
I added this validation because api key is required value to set. For some reason it was missing from this helm chart.

https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/ns1.md